libtransmission/CMakeLists.txt

```diff
@@ -10,9 +10,11 @@ endif()
 
 check_symbol_exists(SO_REUSEPORT "sys/types.h;sys/socket.h" HAVE_SO_REUSEPORT)
 
-add_compile_options(
-    # equivalent of XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC YES for this directory
-    $<$<AND:$<BOOL:${APPLE}>,$<COMPILE_LANGUAGE:C,CXX>>:-fobjc-arc>)
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Comment:
This starts to get messy. I think it'd be better to test if a compiler supports the flag or not (see `check_c_compiler_flag()` and `check_cxx_compiler_flag()` in top-level CMakeLists.txt) and use it based on that. Maybe even base the below `tr_allow_compile_if()` conditions on that too.

Reply:
Done in c1b36fd83bf1a89b2d8e4d0b376fc71117b1698c

Reply:
Actually, utils.cc won't know itself if we're on a compiler supporting ARC or not, so it can only build against `defined(__clang__)`. So let's keep the ARC feature clearly for Clang only, without relying on `check_cxx_compiler_flag`, which itself is a risky call since it can produce false-positives easily due to its underlying implementation.

Reply:
So, mea culpa for me taking a whole month to come back to this build issue, but my latest version is now on 0da70b1077d52f7dbd35bdff20eb9d57284eb10b

Reply:
@Coeur Great, so I can close this PR now, right?

Reply:
It still needs to be reviewed, approved and merged. :)

Reply:
Got it.